src/pages/place/[placeId]/status.js

```diff
@@ -1,48 +1,148 @@
 import { useRouter } from 'next/router';
 import { QueryStatus } from '../../../hooks/api/authenticated-query';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 import { getApplicationServerSideProps } from '../../../utils/getApplicationServerSideProps';
 import { Alert, AlertVariants } from '../../../components/publiq-ui/Alert';
 import { Button, ButtonVariants } from '../../../components/publiq-ui/Button';
 import { Spinner } from '../../../components/publiq-ui/Spinner';
+import { RadioButtonGroup } from '../../../components/publiq-ui/RadioButtonGroup';
 import { Page } from '../../../components/publiq-ui/Page';
+import { TextAreaWithLabel } from '../../../components/publiq-ui/TextAreaWithLabel';
 import { useTranslation } from 'react-i18next';
-import { useGetPlaceById } from '../../../hooks/api/places';
+import { useChangeStatus, useGetPlaceById } from '../../../hooks/api/places';
+import { Text } from '../../../components/publiq-ui/Text';
+import { getValueFromTheme } from '../../../components/publiq-ui/theme';
+import { Stack } from '../../../components/publiq-ui/Stack';
+import { Inline } from '../../../components/publiq-ui/Inline';
+import { useIsClient } from '../../../hooks/useIsClient';
+
+const getValue = getValueFromTheme('statusPage');
+
+const OfferStatus = {
+  AVAILABLE: 'Available',
+  TEMPORARILY_UNAVAILABLE: 'TemporarilyUnavailable',
+  UNAVAILABLE: 'Unavailable',
+};
+
+const maxLengthReason = 200;
 
 const Status = () => {
   const { t, i18n } = useTranslation();
   const router = useRouter();
+  const isClient = useIsClient();
   const { placeId } = router.query;
 
   const [errorMessage, setErrorMessage] = useState();
+  const [type, setType] = useState('');
+  const [reason, setReason] = useState('');
 
-  const handleError = (error) => {
-    setErrorMessage(error.message);
-  };
-  const { data: place = {}, status } = useGetPlaceById(
+  const handleError = (error) => setErrorMessage(error.message);
+  const { data: place = {}, status: queryStatus } = useGetPlaceById(
     { id: placeId },
     { onError: handleError },
   );
+  const handleSuccessChangeStatus = () => {
+    if (isClient) {
+      router.push(`/place/${placeId}/preview`);
+    }
```

Comment:
Is the isClient check needed here? You will never trigger this handler from the server